bindings/matrix-sdk-ffi/src/session_store.rs

```diff
@@ -0,0 +1,222 @@
+use std::sync::Arc;
+#[cfg(feature = "sqlite")]
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+#[cfg(feature = "sqlite")]
+use matrix_sdk::SqliteStoreConfig;
+use tracing::debug;
+use zeroize::Zeroizing;
+
+use crate::{client_builder::ClientBuildError, helpers::unwrap_or_clone_arc};
+
+/// The store paths the client will use when built.
+#[derive(Clone)]
+struct SessionPaths {
+    /// The path that the client will use to store its data.
+    data_path: String,
+    /// The path that the client will use to store its caches. This path can be
+    /// the same as the data path if you prefer to keep everything in one place.
+    cache_path: String,
+}
+
+/// A builder for configuring a Sqlite session store.
+#[cfg(feature = "sqlite")]
+#[derive(Clone, uniffi::Object)]
+pub struct SqliteSessionStoreBuilder {
+    paths: SessionPaths,
+    passphrase: Zeroizing<Option<String>>,
+    pool_max_size: Option<usize>,
+    cache_size: Option<u32>,
+    journal_size_limit: Option<u32>,
+    system_is_memory_constrained: bool,
+}
+
+#[cfg(feature = "sqlite")]
+#[matrix_sdk_ffi_macros::export]
+impl SqliteSessionStoreBuilder {
+    /// Construct a SqliteSessionStoreBuilder and set the paths that the client
+    /// will use to store its data and caches.
+    ///
+    /// Both paths **must** be unique per session as the SDK stores aren't
+    /// capable of handling multiple users, however it is valid to use the
+    /// same path for both stores on a single session.
+    #[uniffi::constructor]
+    pub fn new(data_path: String, cache_path: String) -> Self {
+        Self {
+            paths: SessionPaths { data_path, cache_path },
+            passphrase: Zeroizing::new(None),
+            pool_max_size: None,
+            cache_size: None,
+            journal_size_limit: None,
+            system_is_memory_constrained: false,
+        }
+    }
+
+    /// Set the passphrase for the stores given to
+    /// [`ClientBuilder::paths`].
+    pub fn passphrase(self: Arc<Self>, passphrase: Option<String>) -> Arc<Self> {
+        let mut builder = unwrap_or_clone_arc(self);
+        builder.passphrase = Zeroizing::new(passphrase);
+        Arc::new(builder)
+    }
+
+    /// Set the pool max size for the SQLite stores given to
+    /// [`ClientBuilder::session_paths`].
```

Comment:
Same here. You've also forgot to rename the method.

```suggestion
    /// [`SqliteSessionStoreBuilder::paths`].
```